### ISSUE TITLE
Button use="link" ломает верстку в инлайновом контексте с переносами

### ISSUE THREAD

**Issue by d-hand** [skbkontur/retail-ui#1728]
`<Button` use="link" />` ломает верстку когда его используешь в инлайновом контексте с переносами.

Есть такой компонент, который раскрывается/скрывается по клику 

![1](https://user-images.githubusercontent.com/4019144/68639524-a0a0a480-0526-11ea-8c72-6b20ab07abdb.PNG)

При попытке использовать `<Button use="link" />` ломается верстка

![2](https://user-images.githubusercontent.com/4019144/68639720-405e3280-0527-11ea-8d93-6564bdb86449.PNG)

Пытался обернуть текст в кнопке и навесить класс `white-space: normal;`

![3](https://user-images.githubusercontent.com/4019144/68640265-ff671d80-0528-11ea-84a3-acd8774a30cc.PNG)

Перенос починился, но выравнивание по центру и все ломает. 
`<Button align={"left"} ... />` не помогает.

- Библиотека @skbkontur/react-ui": "1.3.0"
- Браузер chrome 71 
- ОС Windows 10;

Кажется, что `<Button use="link" />` должен вести себя точно также как `<Link />

**Comment by wKich** [comment]
В компоненте `Button` для класса `caption` указан стиль `white-space: nowrap;`, а выравнивание по центру происходит из-за стиля `text-align: center;` в классе `root`.

Кажется что ничего не мешает нам для кнопки-ссылки поменять `white-space` и `text-align` на соответствующие тому как должны отображаться ссылки.

Нужно иметь в виду что если мы изменим для кнопки с `use="link"` выравнивание по левому краю, что с точки зрения логики является более правильным, так как именно таким образом должны вести себя ссылки. То мы сломаем верстку в проектах, которые каким-то образом завязались на выравнивание таких кнопок по центру. Нужно не забыть отразить этот в Changelog

**Comment by dzekh** [comment]
Да, она по идее тут инлайн элементом должна становиться, и ей тогда, так же как инлайн элементу нельзя будет задать определенную ширину, и она должна будет наследовать параметры текста в который она вставляется — если текст выровнен по левому краю — она по левому, если по центру — она по центру. 

Может сделаем новый проп use="inline"?

**Comment by lossir** [comment]
Мне кажется ошибочным, что свойство `use` содержит значение, влияние которого серьёзно отличается от всех других значений.
Обсуждали устно с @wKich, он предложил, как вариант, добавить отдельный проп:
```jsx
<Button link />
```

**Comment by dzekh** [comment]
@d-hand Еще актуально? Вы как-то вышли из этой ситуации?

**Comment by dzekh** [comment]
Можно закрыть?

**Comment by zhzz** [comment]
Мне кажется мы так и не определились, что такое все-таки этот `<Button use="link" />`:

- либо это кнопка, которая просто выглядит как ссылка, но не наследует размер шрифта и не переносит текст внутри себя;
- либо это ссылка, которая ведет себя как настоящая ссылка, просто с элементом `<button />` внутри для семантики;

Второй кейс выглядит так, как будто бы он должен реализовываться в компоненте `<Link />`, а не `<Button />`.

**Comment by d-hand** [comment]
Ох... Это так давно было, что я забыл про эту багу... Чтобы обойти проблему мы стали использоваться Link вместо Button, но кажется, это неправильно, потому что нет href. Поэтому создали эту багу.

**Comment by zhzz** [comment]
@JackUait, спасибо за погружение в проблему и стремление помочь!

Есть еще несколько нюансов, которые следует учесть:

1. У компонента \<Link /> помимо href есть еще проп onClick. Т.е. он может быть интерактивным в отсутствии href.
2. По моему мнению, \<Button use="link" /> не должен копировать поведение обычной ссылки, только внешний вид. А такие вещи как размеры, отступы, выравнивание и перенос текста должно сохраняться от Button. Иначе какой смысл использовать именно Button? В том же пункте [6](https://guides.kontur.ru/components/hyperlink/#06) есть пометка про кликабельную область ссылки-кнопки. Но тут можно поспорить. Нужно понять, для чего вообще создавался этот use="link" и когда он нужен.
3. Гайды все-таки допускают использование ссылок для запуска неосновных действий. И иногда при этом требуется сохранить inline-поведение ссылки и наследование стиля текста (как в этом issue). Для этого можно предусмотреть использование тега \<button /> внутри компонента \<Link />, чтобы удовлетворить стилям и семантике.

Т.е. я лично склоняюсь к тому, чтобы различать \<Button use="link" /> и \<Link />. Можно добавить в каждый что необходимо и использовать по назначению. Но думаю, что Button должен всегда оставаться инлайн-блочным с размерами и прочим, а Link - инлайновым.

**Comment by JackUait** [comment]
> Иначе какой смысл использовать именно Button?

На уровне семантики тег `<button/>` предназначен для обработки события клика, а тег `<a/>` предназначен для перехода на ресурс указанный в `href`, поэтому браузеры, вспомогательные устройства и поисковые системы различают их по-разному.
1. Если в тег `<a/>` не передан `href` это вызовет проблемы у пользователей пользующихся скрин-ридерами.
2. Поисковые системы не любят ссылки, которые никуда не ведут и расценивают большое количество пустых ссылок (так как пустые ссылки рендерятся в обычный текст) на странице как нечестный способ SEO продвижения и наоборот занижают такую страницу в поисковой выдаче.

> У компонента <Link /> помимо href есть еще проп onClick.

Не уверен, что это пример хорошего дизайна, из-за причин описанных выше. 
Не вижу ничего плохого в том, чтобы ссылка обрабатывала дополнительное действие (особенно в контексте `SPA`/`SSR`), но только в том случае, если это действие _дополнительное_, а не основное, потому что основное действие у ссылки - это переход по адресу указанному в `href`, [иначе это не ссылка, а кнопка](https://www.youtube.com/watch?v=KAK-WAb9vow&t=837s). 
То есть при попытке использовать контрол `Link` в качестве кнопки должна быть выброшена ошибка сообщающая о том, что для использования `onClick` в контрол `Link` также должен быть передан проп `href`, иначе (если нет ссылки, которую можно прокинуть) предлагать использовать контрол `Button`.

Итоговое сообщение об ошибке может звучать так:
> Чтобы воспользоваться пропом `onClick` в компоненте `Link`, вам также необходимо передать ссылку в проп `href`. Если пользователь при клике на элементе не должен переходить по ссылке, используйте компонент `Button`, вместо компонента `Link`.
<br/>

> Для этого можно предусмотреть использование тега `<button />` внутри компонента `<Link />`, чтобы удовлетворить стилям и семантике.

На первый взгляд это хорошее решение, но тег `<button/>` [не может быть вложен](https://stackoverflow.com/a/6393863) в тег `<a/>`.
Если отправить код ниже на проверку в [валидатор](https://validator.w3.org/nu/#textarea), то мы получим ошибку: **"The element button must not appear as a descendant of the a element"**. Если не ошибаюсь, даже `React` предупреждает, что такие штуки незаконны.
```html
 <a href="www.stackoverflow.com">
   <button>simple button</button>
 </a>
```

>  Можно добавить в каждый что не хватает и использовать по назначению.

Да, отличная идея, нужно продумать `API` обоих компонентов, часть нехватающего `API` для контрола `Link` я уже описал в #2594. Намедни подумаю над тем, что можно сделать с `API` контрола `Button` относительно контрола `Link` и наоборот.

> Но думаю, что Button должен всегда оставаться инлайн-блочным с размерами и прочим, а Link - инлайновым.

С этим я не согласен по вышеописанным причинам. Мы не должны опираться на [различия в стилях](https://youtu.be/KAK-WAb9vow?t=874) при выборе контрола, мы должны опираться на их функциональное предназначение. Думаю, что мы наоборот должны проработать кейсы стилизации в новом `API` контролов `Link` и `Button`, чтобы позволить пользователю выбирать контрол в зависимости от предоставляемого функционала, а не от стилей.

**Comment by zhzz** [comment]
Еще [кейс](https://kontur.slack.com/archives/C013HTCE18Q/p1630405142033600) про ширину и text-overflow, которые не работают для use="link".
